lexos/static/js/scripts_statistics.js

```diff
@@ -1,66 +1,247 @@
-/* #### INITIATE SCRIPTS ON $(DOCUMENT).READY() #### */
-$(document).ready(function () {
+/**
+ * The function to run the error modal.
+ * @param htmlMsg {string}: the message to display.
+ */
+function runModal (htmlMsg) {
+  $('#error-modal-message').html(htmlMsg)
+  $('#error-modal').modal()
+}
 
-  // hide the normalize options
-  $("#normalize-options").css({"visibility":"hidden"});
-
-  // set the normalize option to raw count
-  $("#normalizeTypeRaw").attr("checked", true)
+/**
+ * At least one document is required to run the stats.
+ * @returns {string | null}: the errors that is checked by JS, if no error the result will be null.
+ */
+function submissionError () {
+  if ($('#num_active_files').val() < 1) { return 'You must have at least 1 active documents to proceed!' } else { return null }
+}
 
+/**
+ * The function to convert the form into json
+ * @returns {{string: string}}: the form converted to json
+ */
+function jsonifyForm () {
+  const form = {}
+  $.each($('form').serializeArray(), function (i, field) {
+    form[field.name] = field.value || ''
+  })
+  return form
+}
 
-  // Reset the maximum number of documents when a checkbox is clicked
-  $('.minifilepreview').click(function () {
-    $('#cullnumber').attr('max', $('.minifilepreview:checked').length)
+/**
+ * Send the ajax request.
+ * @param url {string}: the url to post.
+ * @param form {{string: string}}: the form data packed into an object.
+ * @returns {jQuery.Ajax}: an jQuery Ajax object.
+ */
+function sendAjaxRequest (url, form) {
+  return $.ajax({
+    type: 'POST',
+    url: url,
+    contentType: 'application/json; charset=utf-8',
+    data: JSON.stringify(form)
   })
+}
+
+/**
+ * Format the ajax call response to HTML format string.
+ * @param response {json}: a json format string.
+ * @return {string}: formatted file report.
+ */
+function formatFileReportResponse (response) {
+  // Extract constant result from response.
+  const token_name = response['token_name']
+  const mean = `<p>Average document size is ${response['mean']} ${token_name}</p>`
+  const std_deviation = `<p>Standard deviation of documents is ${response['std_deviation']} ${token_name}</p>`
+  const inter_quartile_range = `<p>Inter quartile range of documents is ${response['inter_quartile_range']} ${token_name}</p>`
+  // Extract standard deviation anomaly information.
+  let anomaly_se
```

Comment:
I understand this code is a pain. However, because of the change of data structure in the backend, you probably need to do more experiments on this code.